src/main/java/com/minecolonies/coremod/util/BarbarianUtils.java

```diff
@@ -0,0 +1,228 @@
+     * @param numberOfSpawns The number of times the entity should be spawned
+     * @param spawnLocation  the location at which to spawn the entity
+     * @param world          the world in which the colony and entity are
+     */
+    public static void spawn(final ResourceLocation entityToSpawn, final int numberOfSpawns, final BlockPos spawnLocation, final World world)
+    {
+
+        if (spawnLocation != null && entityToSpawn != null && world != null)
+        {
+            final int x = spawnLocation.getX();
+            final int y = spawnLocation.getY();
+            final int z = spawnLocation.getZ();
+
+            IntStream.range(0, numberOfSpawns).forEach(theInteger ->
+            {
+                final Entity entity = EntityList.createEntityByIDFromName(entityToSpawn, world);
+
+                if (entity != null)
+                {
+                    setBarbarianItems(entityToSpawn, entity);
+                    entity.setLocationAndAngles(x, y, z, MathHelper.wrapDegrees(world.rand.nextFloat() * WHOLE_CIRCLE), 0.0F);
+                    world.spawnEntity(entity);
+                }
+            });
+        }
+    }
+
+    /**
+     * Sets the various items for each type of barbarian
+     *
+     * @param entityToSpawn The entity which to act upon
+     */
+    private static void setBarbarianItems(ResourceLocation entityToSpawn, Entity entity)
+    {
+        if (entityToSpawn.equals(barbarian))
+        {
+            entity.setItemStackToSlot(EntityEquipmentSlot.MAINHAND, new ItemStack(Items.STONE_AXE));
+        }
+        if (entityToSpawn.equals(archer))
+        {
+            entity.setItemStackToSlot(EntityEquipmentSlot.MAINHAND, new ItemStack(Items.BOW));
+        }
+        if (entityToSpawn.equals(chief))
+        {
+            entity.setItemStackToSlot(EntityEquipmentSlot.MAINHAND, new ItemStack(Items.GOLDEN_SWORD));
+            entity.setItemStackToSlot(EntityEquipmentSlot.CHEST, new ItemStack(Items.CHAINMAIL_CHESTPLATE));
+            entity.setItemStackToSlot(EntityEquipmentSlot.LEGS, new ItemStack(Items.CHAINMAIL_LEGGINGS));
+            entity.setItemStackToSlot(EntityEquipmentSlot.FEET, new ItemStack(Items.CHAINMAIL_BOOTS));
+        }
+    }
+
+    /**
+     * Calculate a random spawnpoint along the colony's border
+     *
+     * @param theWorld in the world.
+     * @param colony   the Colony to spawn the barbarians near.
+     * @return Returns the random blockPos
+     */
+    private static BlockPos calculateSpawnLocation(final World theWorld, final Colony colony)
+    {
+        final ColonyView colonyView = ColonyManager.getClosestColonyView(theWorld, colony.getCenter());
+        if (colonyView == null)
+        {
+            return null;
+        }
+        final BlockPos center = colonyView.getCenter();
+        final int radius = Configurations.workingRangeTownHall;
+
+        final int RandomDegree = theWorld.rand.nextInt(NUMBER_OF_POSSIBLE_CASES);
+
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00117)